bbn-test-5/finality-provider/upgrades/v2rc0/README.md

```diff
@@ -0,0 +1,190 @@
+tightly coupled with the Babylon Genesis Testnet software upgrade.
+
+The following steps must be executed **before the Babylon Genesis Testnet
+upgrade height is reached**, to ensure that your finality provider does not
+experience downtime or submit invalid signatures:
+1. **Stop the Finality Provider and EOTS daemons**.
+2. **Swap the Babylon Genesis finality provider and EOTS daemon binaries with
+   the new ones**.
+3. In your Babylon Genesis Finality Provider config, append the following
+   configuration to the `[Application Options]` section:
+   ```shell
+   [Application Options]
+
+   ; The height at which the context signing will start
+   ContextSigningHeight = X - 1
+   ```
+4. **Start the Finality Provider and the EOTS Daemons.**
+
+After these steps are completed, verify your Finality Provider is signing blocks
+following the steps [here](#313-verification) and wait until the Babylon Genesis
+Testnet upgrade height `1692200` is reached.
+
+#### 3.1.3. Verification
+
+Verify that your Babylon Finality Provider is voting as expected:
+  - Query the Babylon node directly, replacing `FP_BTC_PK_HEX` with the BTC
+    public key of your Babylon Finality Provider in hex format. The height
+    should be growing.
+    ```shell
+    babylond q btcstaking finality-provider \
+      FP_BTC_PK_HEX \
+      --node https://babylon-testnet-rpc.polkachu.com:443 -o json \
+      | jq -r .finality_provider.highest_voted_height
+    ```
+  - Check your Babylon Finality Provider logs, and look for logs of the
+    following format:
+    ```shell
+    LOG_TIMESTAMP	info	successfully submitted the finality signature to the consumer chain	{"consumer_id": "bbn-test-5", "pk": "FP_BTC_PK_HEX", "start_height": X, "end_height": X, "tx_hash": "TX_HASH"}
+    ```
+  - A network explorer can also be consulted (examples:
+   [Xangle](https://babylon-explorer.xangle.io/testnet/finality-providers),
+   [Nodes.guru](https://testnet.babylon.explorers.guru/finality-providers),
+   [Mintscan](https://www.mintscan.io/babylon-testnet/finality-providers)).
+   Explorers have a data indexing overhead, so it's likely that your finality
+   signatures will be reflected after ~1 minute.
+
+### 3.2. Upgrade the Babylon node
+
+#### 3.2.1. Preparation
+
+Prepare for the Babylon `v3.0.0-rc.0` node upgrade following the instructions
+[here](../../../babylon-node/upgrades/v3/README.md#3-1-preparation).
+
+#### 3.2.2. Babylon Node Upgrade
+
+Once the Babylon Genesis Testnet upgrade height `1692200` is reached, perform the
+following steps:
+1. **Upgrade your Babylon node to `v3.0.0-rc.0` following the instructions
+   [here](../../../babylon-node/upgrades/v3/README.md#3-2-execution).**
+2. Wait until the Babylon produces block `1692200`. This can take up to 5 minutes,
+   depending on how fast Babylon CometBFT Validators upgrade their Babylon
+   nodes.
+
+#### 3.2.3. Verification
+
+After completing the [execution](#322-execution) section, perform the following
```

Comment:
`After completing the [execution](#322-execution) section of the Finality Provider upgrade, perform the following`